precompiles/staking/types.go

```diff
@@ -49,6 +55,96 @@ type EventCancelUnbonding struct {
 	CreationHeight   *big.Int
 }
 
+// NewMsgCreateValidator creates a new MsgCreateValidator instance and does sanity checks
+// on the given arguments before populating the message.
+func NewMsgCreateValidator(args []interface{}, denom string) (*stakingtypes.MsgCreateValidator, common.Address, error) {
+	if len(args) != 7 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 7, len(args))
+	}
+
+	description, ok := args[0].(struct {
+		Moniker         string "json:\"moniker\""
+		Identity        string "json:\"identity\""
+		Website         string "json:\"website\""
+		SecurityContact string "json:\"securityContact\""
+		Details         string "json:\"details\""
+	})
```

Comment:
define this as a custom struct at the top of the file

Reply:
done!